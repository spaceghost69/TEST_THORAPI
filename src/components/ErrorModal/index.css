.errorDialog{
    background-color:rgba(253,126,20,.75);
}

.errorDialogMessage{
    overflow-x:scroll;
}