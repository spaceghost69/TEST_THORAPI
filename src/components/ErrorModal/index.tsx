import React from "react";

import { Modal } from "react-bootstrap";
import { FiAlertTriangle } from "react-icons/fi";
import CoolButton from "../CoolButton";

interface ErrorModalProps {
  errorMessage: any;
  callback?: any;
}

const ErrorModal: React.FC<ErrorModalProps> = ({ errorMessage, callback }) => {
  return (
    <Modal
      show={true}
      className={"errorDialog"}
      title="Error"
      size="sm"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header>
        <Modal.Title id="contained-modal-title-vcenter">
          <FiAlertTriangle size={28} color="orange" /> Error
        </Modal.Title>
      </Modal.Header>
      <Modal.Body className={"errorDialogMessage"}>{errorMessage}</Modal.Body>
      <Modal.Footer>
        <CoolButton
          variant={"info"}
          data-dismiss="alert"
          aria-label="Close"
          onClick={() => callback()}
        >
          Ok
        </CoolButton>
      </Modal.Footer>
    </Modal>
  );
};

export default ErrorModal;
