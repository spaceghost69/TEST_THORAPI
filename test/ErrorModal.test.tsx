import React from "react";
import renderer from "react-test-renderer";
import ErrorModal from "../src/components/ErrorModal";

describe("Snapshot testing of ErrorModal", () => {
  it("ErrorModal snapshot", () => {
    const ErrorModalT = renderer
      .create(
        <ErrorModal
          show={true}
          onHide={()=>{}}
        />
      )
      .toJSON();
    expect(ErrorModalT).toMatchSnapshot();
  });
});