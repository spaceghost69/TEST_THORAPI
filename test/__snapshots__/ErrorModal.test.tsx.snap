// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing of ErrorModal ErrorModal snapshot 1`] = `null`;
